database/user_service.go

```diff
@@ -0,0 +1,178 @@
+package database
+
+import (
+	"context"
+	"errors"
+	"favourites/middleware"
+	"favourites/models"
+	"favourites/utils"
+	"fmt"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"time"
+)
+
+type UserService interface {
+	GetAll(ctx context.Context) ([]models.User, error)
+	GetByID(ctx context.Context, id string) (models.User, error)
+	Create(ctx context.Context, m *models.User) error
+	CreateAll(ctx context.Context, m []*models.User) error
+	Update(ctx context.Context, id string, m models.User) (int64, error)
```

Comment:
In the Create and CreateAll methods, your User inputs are pointers, but not in the Update one. Is there any specific reason for that??

Reply:
Well, the thing is, I forgot to use and update these, so there is no reason for it; it should have been a pointer. 😅